plugins/fishlim/tests/fish.c

```diff
@@ -0,0 +1,195 @@
+/*
```

Comment:
Are you literally copying an entire file here? Don't do that...

Reply:
It is the original implementation, for uses in unit tests later

Reply:
Not acceptable. Single shared source between tests and plugin.

Reply:
Single shared source mean add old implementation with new, I think that is not good, what do you think about to resolve this?

Reply:
I'm not sure I understand the problem.

Reply:
I close this, tests implemented.